app/view_models/prior_authority/v1/quote.rb

```diff
@@ -53,10 +53,26 @@ def travel_cost_per_unit
         NumberTo.pounds(travel_cost_per_hour)
       end
 
+      def formatted_travel_cost_per_unit
+        I18n.t(
+          'per_hour',
+          gbp: travel_cost_per_unit,
+          scope: 'prior_authority.application_details.items.per_unit_descriptions'
+        )
+      end
+
       def base_cost_per_unit
         NumberTo.pounds(cost_type == 'per_item' ? cost_per_item : cost_per_hour)
       end
 
+      def formatted_base_cost_per_unit
```

Comment:
If the `quote#item_type` is not `"item"` but instead something like `"word"`, then I believe this method should return something like "£10.00 per word" instead of "£10.00 per item"

Reply:
i can get this working in a smelly way but without a matching `cost_per...` method we  fall back to using `cost_per_hour` at line#65 for the value so you would get "£#{cost_per_hour} per word"


```ruby
def formatted_base_cost_per_unit
  scope = 'prior_authority.application_details.items.per_unit_descriptions'

  if I18n.exists?(cost_type, scope:)
    I18n.t(cost_type, gbp: base_cost_per_unit, unit_description: cost_type, scope:)
  else
    I18n.t('other', gbp: base_cost_per_unit, unit_description: cost_type, scope:)
  end
end
```

```yml
per_unit_descriptions:
  per_item: "%{gbp} per item"
  per_hour: "%{gbp} per hour"
  other: "%{gbp} per %{unit_description}"
```

unless we have a `cost_per_other` capability to fallback on and whose key/value is being sent in the JSON it feels dubious. Could get more "meta" or extract to a separate class to handle the logic it but it feels like handling an edge case that would break in other ways at the moment.

what do you think?

Reply:
So don't worry, it's not as complicated as you think. `cost_type` will only _ever_ be `per_item` or `per_hour`. It's just that if cost_type is `per_item`, we need to refer to `item_type`, which will be one of `item`, `minute`, `word` or `page`.